SimPEG/InvProblem.py

```diff
@@ -104,7 +129,15 @@ def getFields(self, m, store=False, deleteWarmstart=True):
                 break
 
         if f is None:
-            f = self.prob.fields(m)
+            if isinstance(self.dmisfit, DataMisfit.BaseDataMisfit):
+                f = self.dmisfit.prob.fields(m)
+            elif isinstance(self.dmisfit, ObjectiveFunction.BaseObjectiveFunction):
```

Comment:
I see this more as a patch for now, but some of these ideas can stick around when we redefine the InvProb (or remove it). We do want to be able to store the fields, so I have added a `_hasFields` flag to the `BaseObjectiveFunction`, and the fields will be created and stored in a list on the `invProb`. Thoughts @rowanc1?